datasketches/src/frequencies/sketch.rs

```diff
@@ -0,0 +1,408 @@
+
+    /// Returns the current map capacity.
+    pub fn get_current_map_capacity(&self) -> usize {
+        self.cur_map_cap
+    }
+
+    /// Returns the configured lg_max_map_size.
+    pub fn get_lg_max_map_size(&self) -> u8 {
+        self.lg_max_map_size
+    }
+
+    /// Returns the current map size in log2.
+    pub fn get_lg_cur_map_size(&self) -> u8 {
+        self.hash_map.get_lg_length()
+    }
+
+    /// Updates the sketch with a count of one.
+    pub fn update(&mut self, item: T) {
+        self.update_with_count(item, 1);
+    }
+
+    /// Updates the sketch with an item and count.
+    pub fn update_with_count(&mut self, item: T, count: i64) {
+        if count == 0 {
+            return;
+        }
+        assert!(count > 0, "count may not be negative");
+        self.stream_weight += count;
+        self.hash_map.adjust_or_put_value(item, count);
+        self.maybe_resize_or_purge();
+    }
+
+    /// Merges another sketch into this one.
+    pub fn merge(&mut self, other: &Self)
+    where
+        T: Clone,
+    {
+        if other.is_empty() {
+            return;
+        }
+        let merged_total = self.stream_weight + other.stream_weight;
+        for (item, count) in other.hash_map.iter() {
+            self.update_with_count(item.clone(), count);
+        }
+        self.offset += other.offset;
+        self.stream_weight = merged_total;
+    }
+
+    /// Resets the sketch to an empty state.
+    pub fn reset(&mut self) {
+        *self = Self::with_lg_map_sizes(self.lg_max_map_size, LG_MIN_MAP_SIZE);
+    }
+
+    /// Returns frequent items using the sketch maximum error as threshold.
+    pub fn get_frequent_items(&self, error_type: ErrorType) -> Vec<Row<T>>
+    where
+        T: Clone,
+    {
+        self.get_frequent_items_with_threshold(error_type, self.offset)
+    }
+
+    /// Returns frequent items using a custom threshold.
+    pub fn get_frequent_items_with_threshold(
+        &self,
+        error_type: ErrorType,
+        threshold: i64,
+    ) -> Vec<Row<T>>
+    where
+        T: Clone,
+    {
+        let threshold = threshold.max(self.offset);
+        let mut rows = Vec::new();
+        for (item, count) in self.hash_map.iter() {
+            let lower = count;
+            let upper = count + self.offset;
+            let include = match error_type {
+                ErrorType::NoFalseNegatives => upper > threshold,
+                ErrorType::NoFalsePositives => lower > threshold,
+            };
+            if include {
+                rows.push(Row {
+                    item: item.clone(),
+                    estimate: upper,
+                    upper_bound: upper,
+                    lower_bound: lower,
+                });
+            }
+        }
+        rows.sort_by(|a, b| b.estimate.cmp(&a.estimate));
+        rows
+    }
+
+    /// Serializes this sketch into a byte vector using the provided serializer.
+    pub fn serialize_with<S: ItemsSerde<T>>(&self, serde: &S) -> Vec<u8>
```

Comment:
```suggestion
    pub fn serialize<S: ItemsSerde<T>>(&self, serde: &S) -> Vec<u8>
```

Other sketches use `serialize` with context. Let's follow this pattern now.